Northstar.Custom/mod/scripts/vscripts/sh_northstar_http_requests.gnut

```diff
@@ -0,0 +1,222 @@
+    table<string, array<string> > headers
+    /** Query parameters for this http request. */
+    table<string, array<string> > queryParameters
+    /** The content type of this http request. Defaults to application/json & UTF-8 charset. */
+    string contentType = "application/json; charset=utf-8"
+    /** The body of this http request. If set, will override queryParameters.*/
+    string body
+    /** The timeout for the http request in seconds. Must be between 1 and 60. */
+    int timeout = 60
+    /** If set, the override to use for the User-Agent header. */
+    string userAgent
+}
+
+global struct HttpRequestResponse
+{
+    /** The status code returned by the remote the call was made to. */
+    int statusCode
+    /** The body of the response. */
+    string body
+    /** The raw headers returned by the remote. */
+    string rawHeaders
+    /** A key -> values table of headers returned by the remote. */
+    table<string, array<string> > headers
+}
+
+struct HttpRequestCallbacks
+{
+    /** 
+     * The function to call if the HTTP request was a success.
+     * Passes in the response received from the remote.
+     */
+    void functionref( HttpRequestResponse ) onSuccess
+
+    /**
+     * The function to call if the HTTP request failed.
+     * int errorCode, string errorMessage.
+     */
+    void functionref( int, string ) onFailure
+}
+
+table< int, HttpRequestCallbacks > pendingCallbacks
+
+/**
+ * Called from native when a HTTP request is successful.
+ * This is internal and shouldn't be used.
+ * Keep in mind that the success can be successful, but have a non-success status code.
+ * @param handle The handle of the request we got a response for. 
+ * @param statusCode The status code returned in the response.
+ * @param body The body returned for GET requests.
+ * @param headers The headers that were returned in the response.
+ */
+void function NSHandleSuccessfulHttpRequest( int handle, int statusCode, string body, string headers )
+{
+    if ( !( handle in pendingCallbacks ) )
+    {
+        return
+    }
+
+    if ( pendingCallbacks[ handle ].onSuccess != null )
+    {
+        HttpRequestResponse response
+        response.statusCode = statusCode
+        response.body = body
+        response.rawHeaders = headers
+
+        // Parse the raw headers into key -> values
+        array<string> values = split( headers, "\n" )
+
+        foreach ( string header in values )
+        {
+            var index = header.find( ":" )
+            if ( index == null )
+            {
+                continue;
+            }
+
+            string name = strip( header.slice( 0, expect int(index) ) )
+            string value = strip( header.slice( expect int(index) + 1 ) );
+
+            if ( name in response.headers )
+            {
+                response.headers[name].append(value)
+            }
+            else
+            {
+                response.headers[name] <- [ value ]
+            }
```

Comment:
```suggestion
            if ( name in response.headers )
                response.headers[name].append(value)
            else
                response.headers[name] <- [ value ]
```

Reply:
No.

Reply:
single line if statements shouldnt have the curly braces